### ISSUE TITLE
5.1.0 changes status exception behaviour

### ISSUE THREAD

**Issue by bjoernmayer** [LogNet/grpc-spring-boot-starter#371]
Hey there,

we recently tried to update from v5.0.0 to v.5.1.0.

Unfortunately, the status exception handling changed behaviour in this release.

At some point in the code we throw an `IllegalArgumentException`. Then catch it, and map it to a StatusException:
```kotlin
fun IllegalArgumentException.toGrpcFailedPreconditionException(): StatusException =
    Status.FAILED_PRECONDITION
        .withCause(this)
        .withDescription("PollData expired")
        .asException()
```

This exception is then thrown in our grpc service.

We would expect this response:
```
ERROR:
  Code: FailedPrecondition
  Message: PollData expired
```

The actual output is:

```
ERROR:
  Code: Internal
  Message: 
```

Wild guess: This one might cause this: https://github.com/LogNet/grpc-spring-boot-starter/pull/344

**Comment by jvmlet** [comment]
Which  log messages do you see from this [class ](https://github.com/LogNet/grpc-spring-boot-starter/blob/master/grpc-spring-boot-starter/src/main/java/org/lognet/springboot/grpc/FailureHandlingSupport.java)?

**Comment by bjoernmayer** [comment]
> Closing call with Status{code=INTERNAL, description=null, cause=null}

**Comment by jvmlet** [comment]
`Closing call with Status{code=INTERNAL, description=null, cause=null}`

This is **THE ONLY** log you see ?

**Comment by bjoernmayer** [comment]
Yep. I even activated debug log level for `org.lognet.springboot.grpc`

**Comment by jvmlet** [comment]
Hmm, I can see the reason for such behavior, but how do you manage to throw exception from grpc method which is not `RuntimeException` ?  Does Kotlin allow to throw exception which was not defined in `throws`  declaration ?

**Comment by jvmlet** [comment]
Please try with latest `5.1.5-SNAPSHOT`

**Comment by jvmlet** [comment]
`5.1.5` was released
